rdmd.d

```diff
@@ -466,24 +457,18 @@ private int rebuild(string root, string fullExe,
         fullExe = fullExe.defaultExtension(".exe");
 
     // Delete the old executable before we start building.
-    yap("stat ", fullExe);
-    if (exists(fullExe))
+    if (Filesystem.exists(fullExe))
```

Comment:
Won't there be two `stat` outputs from `yap` as a consequence of the separate

```D
if (exists)
{
    enforce(isFile)
    ...
}
```

... setup?  Why do we want this to behave differently from all other `exists && isFile` situations?

(The implicit argument here is: why bother with the `existsAsFile`, rather than just allowing separate checks and having chatty output report this accordingly.)

Reply:
I'll change the second one to a `DirectFilesystem` call so it doesn't log.

Reply:
We really shouldn't use `DirectFilesystem` in the main `rdmd` logic.

I think this is fine as it is.

Reply:
ok, changing back, this one isn't a big issue for me, it only happens once so I'm ok with either solution

Reply:
Note that this will get fixed if we implement the rest of #295...which I plan on making a PR for later.

Reply:
Agree that `DirectFilesystem` should not be used from main app logic.

This all becomes much simpler and more consistent if we just forgo `existsAsFile`.  Why not just drop it?

Reply:
We can't in this case.  We have 3 separate paths to execute:

1. When `fullExe` does not exist, do nothing
2. When `fullExe` exists but is a directory, assert
3. When `fullExe` exists and is not a directory, remove it

Reply:
> This all becomes much simpler and more consistent if we just forgo existsAsFile. Why not just drop it?

Ah, I misread you're comment.  You were saying drop `existsAsFile` altogether.  Can you explain you're rationale?

Reply:
If we drop `existsAsFile` altogether then (i) there is no discrepancy in `stat` reports by `rdmd` and `stat` records in e.g. an `strace`, and (ii) we have one less custom use-case to handle.

Put that together, it leads to simpler and more consistent code that is more precise in its `--chatty` output.

Reply:
> If we drop existsAsFile altogether then (i) there is no discrepancy in stat reports by rdmd and stat records in e.g. an strace, and (ii) we have one less custom use-case to handle.

This is not a problem.  If this is your only justification then I think the answer is clear.

Reply:
> This is not a problem. If this is your only justification then I think the answer is clear.

Every line of code that does not need to exist is a maintenance burden.  This method does not _need_ to exist, so why bother implementing it?

Reply:
> Every line of code that does not need to exist is a maintenance burden.

This is most certainly not true.  More lines of code can be easier to maintain than less.

> This method does not need to exist, so why bother implementing it?

Please stop asking me to re-explain this.  You're talking in circles making the same arguments and asking me to repeat mine.